plugins/modules/events_and_notifications_workflow_manager.py

```diff
@@ -0,0 +1,2589 @@
+            'url': webhook_details.get('url'),
+            'method': webhook_details.get('method', 'POST').upper(),
+            'trustCert': webhook_details.get('trust_cert', False),
+            'isProxyRoute': webhook_details.get('is_proxy_route', True)
+        }
+
+        if webhook_details.get('headers'):
+            headers_list = webhook_details['headers']
+            playbook_params['headers'] = []
+            for headers in headers_list:
+                temp_dict = {
+                    'name': headers.get('name'),
+                    'value': headers.get('name'),
+                    'defaultValue': headers.get('default_value'),
+                    'encrypt': headers.get('encrypt')
+                }
+                playbook_params['headers'].append(temp_dict)
+
+        return playbook_params
+
+    def add_webhook_destination(self, webhook_params):
+        """
+        Add or configure REST webhook destination in Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            webhook_params (dict): A dictionary containing the parameters for configuring the REST webhook destination.
+        Returns:
+            self (object): An instance of a class representing the status of the operation, including whether it was
+                successful or failed, any error messages encountered during operation.
+        Description:
+            This function attempts to add a REST webhook destination in Cisco Catalyst Center using the provided parameters.
+            It sends a request to create a webhook destination with the specified parameters.
+            If the operation is successful, it sets the status to "success" and logs a success message.
+            If the operation fails, it sets the status to "failed" and logs an error message along with any error details
+            received from the API response.
+        """
+
+        try:
+            response = self.dnac._exec(
+                family="event_management",
+                function='create_webhook_destination',
+                op_modifies=True,
+                params=webhook_params
+            )
+            self.log("Received API response from 'create_webhook_destination': {0}".format(str(response)), "DEBUG")
+            status = response.get('apiStatus')
+
+            if status == 'SUCCESS':
+                self.status = "success"
+                self.result['changed'] = True
+                self.msg = "Webhook Destination with name '{0}' added successfully in Cisco Catalyst Center".format(webhook_params.get('name'))
+                self.log(self.msg, "INFO")
+                self.result['response'] = self.msg
+                return self
+
+            self.status = "failed"
+            try:
+                failure_msg = response.get('errorMessage').get('errors')
+            except Exception as e:
```

Comment:
Do we really need to handle exception for failed case here? If below looks good, then you can follow this otherwise ignore it. You can follow the below style for other APIs as well if you like it.


```
        if status == 'SUCCESS':
            self.status = "success"
            self.result['changed'] = True
            self.msg = "Webhook Destination with name '{0}' added successfully in Cisco Catalyst Center".format(webhook_params.get('name'))
            self.log(self.msg, "INFO")
            self.result['response'] = self.msg
        else:
            failure_msg = response.get('errorMessage', {}).get('errors') or "Unable to Add Webhook destination with name '{0}' in Cisco Catalyst Center".format(webhook_params.get('name'))
            self.status = "failed"
            self.log(failure_msg, "ERROR")
            self.result['response'] = failure_msg

    except Exception as e:
        self.status = "failed"
        self.msg = "Error while Adding the Webhook destination with the name '{0}' in Cisco Catalyst Center: {1}".format(webhook_params.get('name'), str(e))
        self.log(self.msg, "ERROR")


    return self
```

Reply:
I changes it in below if else format in all places - 
```
 self.status = "failed"
            error_messages = response.get('errorMessage')
            if error_messages:
                failure_msg = error_messages.get('errors')
            else:
                failure_msg = "Unable to Add Webhook destination with name '{0}' in Cisco Catalyst Center".format(webhook_params.get('name'))